### ISSUE TITLE
May selma supports generics ?

### ISSUE THREAD

**Issue by BenoitCharret** [xebia-france/selma#69]
Hi,
In my project, I had some custom mapper. I wanted to harmonized all of them by using generics. But If I don't override the generics method one the interface, selma doesn't find them.

I made an example: https://github.com/BenoitCharret/selma-test/tree/case_4

Let's try to explain:
I define an interface that define how to map generic type:

```java
public interface IListMapper<E,D> {
    E asEntity(D dto);

    D asDto(E entity);
}

When defining custom mapper,  I only want to have : 
@Mapper(
        withCustomFields = {@Field({"src", "field"})}
)
public interface MyListMapper extends IListMapper<SrcString,DstString>{
    
}
````

But, In this way, selma complains about the type of the method : 
Error:java: Failed to generate mapping method for type D to E not supported on com.test.IListMapper.asEntity(D) !
  --> Add a custom mapper or 'withIgnoreFields' on @Mapper or @Maps to fix this ! If you think this a Bug in Selma please report issue here [https://github.com/xebia-france/selma/issues].

If I just override the method, selma compile.

@Mapper(
        withCustomFields = {@Field({"src", "field"})}
)
public interface MyListMapper extends IListMapper<SrcString,DstString>{
    @Override
    SrcString asEntity(DstString dto);

    @Override
    DstString asDto(SrcString entity);
}

I think, the code could be cleaner if we could find those method without override them.

Thanks

**Comment by BenoitCharret** [comment]
This is maybe a duplicate of #63 second comment.

**Comment by slemesle** [comment]
This is now supported ax fixed with #63.
